sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -404,6 +407,13 @@ class ScalaReflectionSuite extends SparkFunSuite {
         StructField("x", IntegerType, nullable = false))), nullable = true))
   }
 
+  test("SPARK-29026: schemaFor for trait without companion constructor throws exception ") {
+    val e = intercept[UnsupportedOperationException] {
+      schemaFor[TraitProductWithoutCompanion]
+    }
+    e.getMessage.contains("Unable to find constructor")
+  }
```

Comment:
^ Scrolling up, an existing test is for `ScroogeLikeExample` where it checks that trait with companion object constructor works.